src/PowerShellEditorServices/Language/FoldingReference.cs

```diff
@@ -60,4 +61,39 @@ public class FoldingReference: IComparable<FoldingReference>
             return string.Compare(this.Kind, that.Kind);
         }
     }
+
+    /// <summary>
+    /// A class that holds a list of FoldingReferences and ensures that when adding a reference that the
+    /// folding rules are obeyed, e.g. Only one fold per start line
+    /// </summary>
+    public class FoldingReferenceList : Dictionary<int, FoldingReference>
+    {
+        /// <summary>
+        /// Adds a FoldingReference to the list and enforces ordering rules e.g. Only one fold per start line
+        /// </summary>
+        public void SafeAdd(FoldingReference item)
+        {
+            if (item == null) { return; }
+
+            // Only add the item if it hasn't been seen before or it's the largest range
+            if (TryGetValue(item.StartLine, out FoldingReference currentItem))
+            {
+                if (currentItem.CompareTo(item) == 1) { this[item.StartLine] = item; }
+            }
+            else
```

Comment:
Can use `return` rather than `else` here

Reply:
This function returns void so the return doesn't seem to be needed.

Reply:
I think he meant:

```csharp
if (references.TryGetValue(item.StartLine, out FoldingReference currentItem))
{
    if (currentItem.CompareTo(item) == 1) { references[item.StartLine] = item; }
    return;
}

references[item.StartLine] = item;
```